coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,390 @@
+        length of a code i.e blank lines can either make code longer, less
+        readable and maintainable or do the reverse.
+        """
+        fix_suggestions = """
+        Following specific rules about the usage of blank lines: using them
+        only when necessary.
+        """
+
+
+@BlankLine.subaspect
+class BlankLineAfterDeclaration:
+    """
+    Blank lines after declaration refer to the blank lines found after
+    declarations.
+    """
+    class docs:
+        example = """
+        int main ()
+        {
+          int a;
+          return 0;
+        }
+        """
+        example_language = 'C'
+        importance_reason = """
+        Having a specific  and reasonable number of blank lines after every
+        block of declarations improves on the readability of the code.
+        """
+        fix_suggestions = """
+        `BlankLintAfterDeclaration` issues can be fixed  specifying (and of
+        course using) a reasonable number of blank lines to use after block
+        declaration.
+        """
+    blank_lines_after_declarations = Taste[int](
+        'Represents the number of blank lines after declarations',
+        (0, 1, 2), default=0)
+
+
+@BlankLine.subaspect
+class BlankLineAfterProcedure:
+    """
+    Blank lines after procedures` refer to the blank lines found after
+    procedures or functions.
+    """
+    class docs:
+        example = """
+        void proc(void){
+            printf("this does nothing");
+        }
+
+        int add(float a, float b){
+            return a + b;
+        }
+        """
+        example_language = 'C++'
+        importance_reason = """
+        Having a specific and reasonable number of blank lines after every
+        procedures improves on the readability of the code.
+        """
+        fix_suggestions = """
+        """
+    blank_lines_after_procedures = Taste[int](
+        'Represents the number of blank lines to use after a procedure or a'
+        ' function', (0, 1, 2), default=1)
+
+
+@BlankLine.subaspect
+class BlankLineAfterClass:
+    """
+    Blank lines after classes refer to the blank lines found after classes'
+    definitions.
+    """
+    class docs:
+        example = """
+        class SomeClass:
+            def __init__(self):
+                raise RuntimeError('Never instantiate this class')
+
+
+        def func():
+            pass
+        """
+        example_language = 'python'
+        importance_reason = """
+        Having a specific number of blank lines after every classes'
+        definitions declarations improves on the readability of the code.
+        """
+        fix_suggestions = """
+        """
+    blank_lines_after_class = Taste[int](
+        'Represents the number of blank lines to use after a class'
+        'definition.', (0, 1, 2, 3), default=2)
+
+
+@BlankLine.subaspect
+class NewlineAtEOF:
+    """
+    Newline at EOF refers to a newline character found at the end of a file.
```

Comment:
this needs a better explanation, people *always* get confused by this. Here is my template when I'm pointing this one out in PRs:

---

Hi,

it looks like you're missing a railing newline before the end of the file here. [This article](http://stackoverflow.com/a/5813359/3212182) has more information about this issue.

---

we should make an effort explaining this really properly IMO